### PR TITLE
ENH: Added MR modality specific attributes and improving usability by using argparse

### PR DESCRIPTION
**Pull request by che85**
- using argparse for providing help and flagged parameters
- if dcmqi is required for an SR it will throw an exception if it was neither found in environment variable `DCMQI_PATH` nor as an additional parameter `-dcmqi <dcmqi binary path>`
- changed indention from 4 to 2 spaces in order to be consistent
- prompting user if output directory does not exist

### CODE REVIEW

File: schema.qdbd [hunk 1]
```diff
@@ -7,9 +7,12 @@
 # common across modalities
 CompositeContext
 -
+InstitutionName
+StationName
+EquipmentUID
+ProductId
 SOPClassUID
 SOPInstanceUID PK
-StudyInstanceUID
```
Comment (fedorov) [inline schema.qdbd]: This should not be removed, it is quite important.

File: tabulate.py [hunk 2]
```diff
@@ -17,32 +20,65 @@
...
-      except:
-        print("Failed to parse: "+dcmName)
+      except (DCMQINotFoundError, KeyError) as exc:
+        print ("Failed to read DICOM %s" % dcmName)
+        if type(exc) is exceptions.KeyError:
+          print ("Missing key: %s " % exc)
+        else:
+          print ("Make sure that you specified dcmqi path either in your environment variable 'DCMQI_PATH' or as an "
+                "additional parameter '-dcmqi <DCMQI binary path>'")
+        print ("Skipping %s" %dcmName)
```
Comment (fedorov) [inline tabulate.py]: I don't think it is a good idea to skip DICOM files when there is any of the attributes missing. They should be set to empty values, but parsing should continue for those files!

**Review by fedorov — CHANGES_REQUESTED** [review]

**Comment by fedorov** [comment]
@che85 what is the status of this PR?

File: tabulate.py (hunk 2 above)
Comment (che85) [reply tabulate.py]: I need to look into this before the PR is ready.

**Comment by che85** [comment]
@fedorov I think https://github.com/QIICR/dcm2tables/pull/5/commits/d90a4fd6d8d68c2ac31c6559e74d5d55c029f47f already fixed your concerns regarding skipping of series. Now that only happens if a SR was found but no dcmqi executable is available. Or should it even in that situation just print a message and continue?

File: DICOMParser.py [hunk 3]
```diff
@@ -55,6 +55,8 @@ def parse(self):
         with open(outputJSON) as jsonFile:
           measurementsJSON = json.load(jsonFile)
           self.readMeasurements(measurementsJSON)
+      else:
```
Comment (fedorov) [inline DICOMParser.py:79]: Since template identifier is optional, can you also add the exception handling?

**Comment by fedorov** [comment]
Yes, I think it should not skip the file. If dcmqi is not available, it still makes sense to extract composite context for the input SRs.

**Comment by che85** [comment]
@fedorov I just removed the "skipping" message. Feel free to test and merge.

File: DICOMParser.py (hunk 3 above)
Comment (che85) [reply DICOMParser.py:79]: I added an exception for that.

**Pull request merged**